book.cmd/src/main/java/com/happystays/book/cmd/infrastructure/BookCommandEventStore.java

```diff
@@ -27,32 +30,43 @@ public class BookCommandEventStore implements EventStore {
-//        eventProducer.produce(baseEvent.getClass().getSimpleName(), baseEvent);
-//        List<EventModel> eventsList = eventStoreRepository.findByAggregateId(aggregateId);
-//        if(expectedVersion != -1 && eventsList.get(eventsList.size() - 1).getVersion() != expectedVersion){
-//            throw new ConcurrencyException();
-//        }
-//        int version = expectedVersion;
-//        BookingSuccessEvent event = (BookingSuccessEvent) baseEvent;
-//        version ++;
-//        event.setVersion(version);
-//        Pnr pnrInformation = event.getPnrList().get(0);
-//        HotelSegment hotelSegmentInformation = pnrInformation.getHotelInfoList().get(0).getHotelSegmentList().get(0);
-//        EventModel eventModel = EventModel.builder().version(version)
-//                .aggregateId(aggregateId)
-//                .aggregateType(BookAggregate.class.getTypeName())
-//                .id(event.getId())
-//                .eventData(PnrEventModel.builder()
-//                        .bookingState(pnrInformation.getStatus())
-//                        .confirmationNumber(hotelSegmentInformation.getConfirmationNumber())
-//                        .beginDate(pnrInformation.getTrip().getBeginDate())
-//                        .endDate(pnrInformation.getTrip().getEndDate())
-//                        .occupancy(hotelSegmentInformation.getOccupancy())
-//                        .totalAmount(pnrInformation.getTrip().getTotalPrice())
-//                        .creationDate(pnrInformation.getCreationDate())
-//                        .build())
-//                .build();
-//            eventStoreRepository.save(eventModel);
+        eventProducer.produce(baseEvent.getClass().getSimpleName(), baseEvent);
+        List<String> confirmationNumbers = new ArrayList<>();
+        Map<String, Object> pnrDataMap = new HashMap<>();
+        List<EventModel> eventsList = eventStoreRepository.findByAggregateId(aggregateId);
+        if (expectedVersion != -1 && eventsList.get(eventsList.size() - 1).getVersion() != expectedVersion) {
+            throw new ConcurrencyException();
+        }
+        int version = expectedVersion;
+        BookingSuccessEvent event = (BookingSuccessEvent) baseEvent;
+        version++;
+        event.setVersion(version);
+        // List of pnr is iterated
+        event.getTrip().getPnrList().stream().forEach(pnr -> {
+            // hotelInfo List is iterated for each pnr
+            pnr.getHotelInfoList().stream().forEach(hotelInfo -> {
+                // hotelSegment for each hotelInfo is iterated
+                hotelInfo.getHotelSegmentList().stream().forEach(hotelSegment -> {
+                    pnrDataMap.put("creationDate", pnr.getCreationDate());
+                    pnrDataMap.put("bookingStatus", pnr.getStatus());
+                    confirmationNumbers.add(hotelSegment.getConfirmationNumber());
+                });
+            });
+        });
```

Comment:
Do we really need this logic.  Creation date can be passed as today's date. status can be take from get(0) . So just stream and collect confirmation number to list , no need of map

Reply:
removed the map and using status, creation date as per suggestion